packages/api/src/hooks/useCryptoTransactions.ts

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useMemo, useState } from 'react';
+import useSubscription from '../useSubscription';
+import { getTruncatedString } from '@deriv/utils';
+
+type TTransaction = NonNullable<
+    NonNullable<ReturnType<typeof useSubscription<'cashier_payments'>>['data']>['cashier_payments']
+>['crypto'][number];
+
+type TStatus = TTransaction['status_code'];
+
+type TDepositStatus = 'PENDING' | 'CONFIRMED' | 'ERROR';
+
+type TWithdrawalStatus = Exclude<TStatus, TDepositStatus>;
+
+// Since BE sends the `status_code` for both `deposit` and `withdrawal` in the same field,
+// Here we modify the BE type to make `status_code` type more specific to the `transaction_type` field.
+type TModifiedTransaction = Omit<TTransaction, 'status_code' | 'transaction_type'> &
+    (
+        | { transaction_type: 'deposit'; status_code: TDepositStatus }
+        | { transaction_type: 'withdrawal'; status_code: TWithdrawalStatus }
+    );
+
+/** A custom hook that returns the list of pending crypto transactions for the current user. */
+const useCryptoTransactions = () => {
+    const { subscribe, data, ...rest } = useSubscription('cashier_payments');
+    const [transactions, setTransactions] = useState<TModifiedTransaction[]>();
+
+    useEffect(() => {
+        setTransactions(old_transactions => {
+            // Get the list of the crypto transactions.
+            const new_transactions = data?.cashier_payments?.crypto as TModifiedTransaction[] | undefined;
+
+            // If there are no new transactions, return the old transactions.
+            if (!new_transactions) return old_transactions;
+
+            // If there are no old transactions, return the new transactions.
+            if (!old_transactions) return new_transactions;
```

Comment:
Quite a lot of comments here. Do they all add that much value. For example, this one is maybe self explanatory?